server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -323,14 +360,24 @@ export class TenantManager {
         newTenantKey: string,
         tenantId: string,
     ): Promise<ITenantKeys> {
+        const lumberProperties = { [BaseTelemetryProperties.tenantId]: tenantId };
         // if key2 is to be refreshed
         if (keyName === KeyName.key2) {
             const decryptedTenantKey1 = this.secretManager.decryptSecret(key1);
             if (decryptedTenantKey1 == null) {
                 winston.error("Tenant key1 decryption failed.");
-                Lumberjack.error("Tenant key1 decryption failed.", { [BaseTelemetryProperties.tenantId]: tenantId });
+                Lumberjack.error("Tenant key1 decryption failed.", lumberProperties);
                 throw new NetworkError(500, "Tenant key1 decryption failed.");
             }
+
+            const cacheKeys = JSON.stringify({
+                key1,
+                key2: this.secretManager.encryptSecret(newTenantKey)
+            });
```

Comment:
Let's create a cacheKey object only here, and move the `JSON.stringify` inside the `setCacheKey` to reduce duplicate code?